src/client.ts

```diff
@@ -108,7 +108,10 @@ export default class Client {
     };
 
     this.context = context;
-    this.statusItem = new StatusItem(this.context, this.ruby);
+    this.statusItems = new StatusItems({
+      context: this.context,
+      ruby: this.ruby,
+    });
```

Comment:
I was thinking that this could be more flexible if we were passing the `client` itself, so that the status item could refer to the current value of `context` and `ruby` at any given point. I guess that means we might need to make (at least the getters of) `ruby` and `context` public.

So, we can do something like [this](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDC0G8CwAoa0AOAnAlgNzABcBTaAfRGIDsBzQgC2gF5oAGAblVXQwFcAjEDmDQaxQtEq0GACgCUibumhZxvLFWgMcEAHQVqdepzTQAvkuz4ipCOMmHZBEL2ILkp9Nr0HpjFs6uJugWKKGolBIiLFTEAO5w8ibAulJGqClpDMzQACwATOxAA):
```ts
class Client {
  private _ruby: Ruby

  public get ruby() {
    return this._ruby;
  }

  private set ruby(ruby: Ruby) {
    this._ruby = ruby;
  }
}

client = new Client();

client.ruby # should work
client.ruby = new Ruby() # should fail
```

Reply:
We can't pass the client itself. TypeScript won't accept a `Client` for `{ context: ..., ruby: ... }`. And we can't import `Client` in `StatusItem` because it causes a circular dependency.

Reply:
It only rejects passing a `client` as `{ context: ..., ruby: ... }` since those getters are currently private on `Client`. If we fix that, it should work.

Reply:
Perfect! I used that and changed refresh to accept the Client interface as well. Much better.